src/components/App.tsx

```diff
@@ -163,14 +164,40 @@ function App() {
   const [nodes, setNodes, onNodesChange] = useNodesState([]);
   const [edges, setEdges, onEdgesChange] = useEdgesState([]);
 
+  const edgeUpdateSuccessful = useRef(true);
+
+  const onEdgeUpdateStart = useCallback(() => {
+    edgeUpdateSuccessful.current = false;
+  }, []);
+
+  const onEdgeUpdate = useCallback((oldEdge: Edge<any>, newConnection: Connection) => {
+    if (getFluxNodeParent(nodes, edges, newConnection.target!) !== undefined) {
+      return;
+    }
```

Comment:
we also need a check for 

```
 isFluxNodeInLineage(nodes, edges, {
        nodeToCheck: connection.target!,
        nodeToGetLineageOf: connection.source!,
      })
```

to prevent recursive loops

Reply:
could maybe extract these two checks into a joint function in `fluxNodes` since we're now using them twice

Reply:
done